lib/instructions/instruction_names.c

```diff
@@ -261,10 +261,800 @@ static const char* names[256] = {
+    "AND",
+    "OR",
+    "XOR",
+    "NOT",
+    "BYTE",
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    "SHA3",
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    "ADDRESS",
+    "BALANCE",
+    "ORIGIN",
+    "CALLER",
+    "CALLVALUE",
+    "CALLDATALOAD",
+    "CALLDATASIZE",
+    "CALLDATACOPY",
+    "CODESIZE",
+    "CODECOPY",
+    "GASPRICE",
+    "EXTCODESIZE",
+    "EXTCODECOPY",
+    NULL,
+    NULL,
+    NULL,
+    "BLOCKHASH",
+    "COINBASE",
+    "TIMESTAMP",
+    "NUMBER",
+    "DIFFICULTY",
+    "GASLIMIT",
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    "POP",
+    "MLOAD",
+    "MSTORE",
+    "MSTORE8",
+    "SLOAD",
+    "SSTORE",
+    "JUMP",
+    "JUMPI",
+    "PC",
+    "MSIZE",
+    "GAS",
+    "JUMPDEST",
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    "PUSH1",
+    "PUSH2",
+    "PUSH3",
+    "PUSH4",
+    "PUSH5",
+    "PUSH6",
+    "PUSH7",
+    "PUSH8",
+    "PUSH9",
+    "PUSH10",
+    "PUSH11",
+    "PUSH12",
+    "PUSH13",
+    "PUSH14",
+    "PUSH15",
+    "PUSH16",
+    "PUSH17",
+    "PUSH18",
+    "PUSH19",
+    "PUSH20",
+    "PUSH21",
+    "PUSH22",
+    "PUSH23",
+    "PUSH24",
+    "PUSH25",
+    "PUSH26",
+    "PUSH27",
+    "PUSH28",
+    "PUSH29",
+    "PUSH30",
+    "PUSH31",
+    "PUSH32",
+    "DUP1",
+    "DUP2",
+    "DUP3",
+    "DUP4",
+    "DUP5",
+    "DUP6",
+    "DUP7",
+    "DUP8",
+    "DUP9",
+    "DUP10",
+    "DUP11",
+    "DUP12",
+    "DUP13",
+    "DUP14",
+    "DUP15",
+    "DUP16",
+    "SWAP1",
+    "SWAP2",
+    "SWAP3",
+    "SWAP4",
+    "SWAP5",
+    "SWAP6",
+    "SWAP7",
+    "SWAP8",
+    "SWAP9",
+    "SWAP10",
+    "SWAP11",
+    "SWAP12",
+    "SWAP13",
+    "SWAP14",
+    "SWAP15",
+    "SWAP16",
+    "LOG0",
+    "LOG1",
+    "LOG2",
+    "LOG3",
+    "LOG4",
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    "CREATE",
+    "CALL",
+    "CALLCODE",
+    "RETURN",
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    "INVALID",
+    "SELFDESTRUCT",
```

Comment:
I think this should be `SUICIDE` in Frontier :)

And technically `INVALID` should only be added at a later hard fork if were are pedantic.

Reply:
The `INVALID` EIP does not have the date specified. I think we can assume that this is a kind of change that affects EVM also back in time.
The similar case is for SELFDESTRUCT (EIP-6) which was not the part of Homestead HF officially.